### PR TITLE
created route for parse general amount of search request

### PR DESCRIPTION
**Pull request by asavienko**
created function thant take auth token from authorized page

### CODE REVIEW

File: api/actions/countFoundCvs.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const querystring = require("querystring");
+const fetch = require("node-fetch");
+
+const countFoundCvs = async ({
```
Comment (bohdanSnappy) [inline api/actions/countFoundCvs.js]:
```suggestion
const gevCvsNumber = async ({
```

File: api/actions/countFoundCvs.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    keyword,
+    regionid,
+    period,
+    searchtype,
+    moveability,
+    sort
+  });
+  const response = await fetch(targetUrl + searchStringRequest);
+  const json = await response.json();
+  const totalCvs = json.Total;
```
Comment (bohdanSnappy) [inline api/actions/countFoundCvs.js]: return json.Total;

File: server.js [hunk 3]
```diff
@@ -5,22 +5,43 @@ const login = require("./api/actions/login");
...
 
 app.get("/cvlist", (req, res) => {
   const query = req.query;
   collection
-    .then(collection => collection.find(query).toArray())
+    .then(collection =>
+      collection
+        .collection("cvs")
+        .find(query)
+        .toArray()
```
Comment (bohdanSnappy) [inline server.js]: check if we need toArray here

File: server.js [hunk 4]
```diff
@@ -29,9 +50,21 @@ app.get("/cvlist", (req, res) => {
       })
     );
 });
-const port = 5000;
-app.listen(port, () => console.log(`Listening on port ${port}...`));
 
+app.post("/cv_preparse", async (req, res) => {
+  const request = req.body;
+  console.log(request);
```
Comment (bohdanSnappy) [inline server.js]: remove console.log

File: server.js [hunk 5]
```diff
@@ -29,9 +50,21 @@ app.get("/cvlist", (req, res) => {
       })
     );
 });
-const port = 5000;
-app.listen(port, () => console.log(`Listening on port ${port}...`));
 
+app.post("/cv_preparse", async (req, res) => {
```
Comment (bohdanSnappy) [inline server.js]: use get instead of post

Comment (bohdanSnappy) [inline server.js]: rename `cv_preparse` to `get-cvs-numbers`

File: src/components/Home/Home.js [hunk 6]
```diff
@@ -1,61 +1,126 @@
...
+const StyledSelect = styled(Select)`
+  width: 150px;
+`;
+const StyledSearch = styled(Search)`
+  padding: 15px;
+`;
+
+function Home({ dictionaryCity, setDictionaryCity }) {
+  const [searchRequest, setSearchRequest] = useState({
+    regionid: 0,
```
Comment (bohdanSnappy) [inline src/components/Home/Home.js]: use camelCase

File: src/components/Home/Home.js [hunk 7]
```diff
@@ -1,61 +1,126 @@
...
+    keywords: ""
+  });
+  const [modalLoading, setModalLoading] = useState(false);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [modalData, setModalData] = useState({});
+
+  useEffect(() => {
+    if (!dictionaryCity.length) {
+      fetch("/dictionary_city")
```
Comment (bohdanSnappy) [inline src/components/Home/Home.js]: move all requests to services/CvService.js

Comment (bohdanSnappy) [inline src/components/Home/Home.js]: const data = await getData();
if (json.confirmation === "fail") {
    return openNotification("error")
}
setDictionaryCity(json.data);

File: src/components/Home/Home.js [hunk 8]
```diff
@@ -1,61 +1,126 @@
...
+  };
+  const onSearchPressed = searchValue => {
+    const editedSearchRequest = searchRequest;
+    editedSearchRequest.keywords = searchValue;
+    setSearchRequest(editedSearchRequest);
+    console.log(searchRequest);
+    setModalLoading(true);
+    setModalVisible(true);
+    fetch("/cv_preparse", {
+      method: "POST",
```
Comment (bohdanSnappy) [inline src/components/Home/Home.js]: move to service

File: src/components/Home/Home.js [hunk 9]
```diff
@@ -1,61 +1,126 @@
...
+            {dictionaryCity.map(record => (
+              <Option value={record.id} key={record.id}>
+                {record.ru}
+              </Option>
+            ))}
+          </StyledSelect>
+        }
+      />
 
+      <Modal
```
Comment (bohdanSnappy) [inline src/components/Home/Home.js]: move modal to separate component

File: src/components/Home/SelectCity.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import { Select } from "antd";
```
Comment (bohdanSnappy) [inline src/components/Home/SelectCity.js]: check if we need this file

File: server.js (hunk 3 above)
Comment (asavienko) [reply server.js]:
```suggestion
collection
    .then(collection =>
      collection
        .collection("cvs")
        .find(query)
        .toArray()
    )
    .then(result => res.json({ confirmation: "success", data: result }))
```

File: src/components/Home/Home.js (hunk 7 above)
Comment (asavienko) [reply src/components/Home/Home.js]:
```suggestion
      async () => {
      if (!dictionaryCity.length) {
        try {
          const response = await getRequest("/dictionary-city");
          if (response.confirmation === "success") {
            setDictionaryCity(response.data);
          } else {
            throw new Error();
          }
        } catch {
          openNotification({
            type: "error",
            message: "Не удалось загрузить списко городов"
          });
        }
      }
    }
```

File: package.json [hunk 11]
```diff
@@ -24,6 +24,7 @@
     "react-router-dom": "^5.0.1",
     "react-scripts": "3.0.1",
     "redux": "^4.0.4",
+    "request": "^2.88.0",
```
Comment (bohdanSnappy) [inline package.json]: check if you need request here

File: server.js [hunk 12]
```diff
@@ -5,22 +5,28 @@ const login = require("./api/actions/login");
...
-  obj.db("rabotaua").collection("cvs")
-);
+const collection = client.then(obj => obj.db("rabotaua"));
 
-app.get("/", (req, res) => {
-  res.send("Hello world...  ");
-});
+const port = 5000;
 
 app.get("/cvlist", (req, res) => {
```
Comment (bohdanSnappy) [inline server.js]: try use async await

File: server.js [hunk 13]
```diff
@@ -29,10 +35,36 @@ app.get("/cvlist", (req, res) => {
...
+        message: err.message
+      })
+    );
+});
+app.get("/total-cvs", async (req, res) => {
+  const page = await initBrowser();
+  const enteredPage = await login(page);
+  const token = await getAuthToken(enteredPage);
+  await enteredPage.close();
+  const query = querystring.stringify(req.query);
```
Comment (bohdanSnappy) [inline server.js]: rename query to queryString

File: src/components/Home/Home.js [hunk 14]
```diff
@@ -1,61 +1,182 @@
...
+  };
+  const onSearchPressed = async searchValue => {
+    const editedSearchRequest = searchRequest;
+    editedSearchRequest.keywords = searchValue;
+    setSearchRequest(editedSearchRequest);
+    setModalLoading(true);
+    setModalVisible(true);
+    try {
+      const response = await getRequest(
+        `/total-cvs?regionid=${searchRequest.regionId}&keywords=${
```
Comment (bohdanSnappy) [inline src/components/Home/Home.js]: move to separate function in services

File: src/components/Home/Home.js [hunk 15]
```diff
@@ -1,61 +1,182 @@
...
+    setModalLoading(false);
+    setModalData(modalBasicData);
+  };
+  const handleModalOk = async () => {
+    try {
+      setOkButtonLoading(true);
+      const response = await getTotalCv(searchRequest);
+      setModalVisible(false);
+      if (response.confirmation === "success") {
+        InformationModal({
```
Comment (bohdanSnappy) [inline src/components/Home/Home.js]:
```suggestion
       return InformationModal({
```

File: src/components/Home/Home.js [hunk 16]
```diff
@@ -1,61 +1,182 @@
...
+                <StyledBoldSpan>{response.minutes}мин.</StyledBoldSpan>
+              </p>
+              <p>Вы можете посмотреть уже отсканированные резюме в списке</p>
+              <p>
+                Когда будут отсканированны все резюме вы получите уведомление
+              </p>
+            </div>
+          )
+        });
+      } else {
```
Comment (bohdanSnappy) [inline src/components/Home/Home.js]:
```suggestion
```

File: src/components/Home/Home.js [hunk 17]
```diff
@@ -1,61 +1,182 @@
...
+        throw new Error();
+      }
```
Comment (bohdanSnappy) [inline src/components/Home/Home.js]:
```suggestion
```

File: src/components/Home/SearchModal.js [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+
+function SearchModal({
+  modalVisible,
+  modalLoading,
+  handleModalCancel,
+  handleModalOk,
+  modalData: { totalCv },
+  okButtonLoading
+}) {
+  const okButtonDisabled = totalCv ? modalLoading : true;
```
Comment (bohdanSnappy) [inline src/components/Home/SearchModal.js]: !totalCv || modalLoading

File: api/actions/connectMongoDb.js [hunk 19]
```diff
@@ -10,4 +10,17 @@ const connectDb = () => {
   return client;
 };
 
-module.exports = connectDb;
+const connectCollection = async (dbName, collectName) => {
+  if (dbName && collectName) {
+    const client = await connectDb();
+    const collection = await client.db(dbName).collection(collectName);
```
Comment (bohdanSnappy) [inline api/actions/connectMongoDb.js]: check if collection name could be an empty string

File: api/actions/connectMongoDb.js [hunk 20]
```diff
@@ -10,4 +10,17 @@ const connectDb = () => {
   return client;
 };
 
-module.exports = connectDb;
+const connectCollection = async (dbName, collectName) => {
```
Comment (bohdanSnappy) [inline api/actions/connectMongoDb.js]: rename collectName to collectionName

**Pull request merged**